File: library/src/main/java/zjm/cst/dhu/library/inject/component/ApplicationComponent.java
package zjm.cst.dhu.library.inject.component;

import android.content.Context;

import dagger.Component;
import zjm.cst.dhu.library.inject.module.ApplicationModule;
import zjm.cst.dhu.library.inject.scope.PerApplication;
import zjm.cst.dhu.library.inject.scope.PerContext;

/**
 * Created by zjm on 2017/5/9.
 */

@PerApplication
@Component(modules = ApplicationModule.class)
public interface ApplicationComponent {

    @PerContext("Application")
    Context getApplicationContext();
}

File: library/src/main/kotlin/zjm/cst/dhu/library/inject/component/ApplicationComponent.kt
package zjm.cst.dhu.library.inject.component

import android.content.Context
import dagger.Component
import zjm.cst.dhu.library.inject.module.ApplicationModule
import zjm.cst.dhu.library.inject.scope.PerApplication
import zjm.cst.dhu.library.inject.scope.PerContext

/**
 * Created by zjm on 2017/5/9.
 */

@PerApplication
@Component(modules = arrayOf(ApplicationModule::class))
interface ApplicationComponent {

    @get:PerContext("Application")
    val applicationContext:
            Context
}
